### PR TITLE
fix(cli): print namespace and mesh for commands

### LINKED ISSUE #2455: Make it clear that CLI uses osm-system as default namespace for various commands
**Please describe the Improvement and/or Feature Request**

The CLI should clearly indicate to users when it is using osm-system as the default namespace
**Scope (please mark with X where applicable)**
- New Functionality          [ ]
- Install                    [ ]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [ ]
- Permissive Traffic Policy  [ ]
- Ingress                    [ ]
- Egress                     [ ]
- Envoy Control Plane        [ ]
- CLI Tool                   [X]
- Metrics                    [ ]
- Certificate Management     [ ]
- Sidecar Injection          [ ]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [ ]
- CI System                  [ ]
- Project Release            [ ]

**Possible use cases**

Users may not be aware that commands such as uninstall and policy use osm-system as the default namespace.

### PR DESCRIPTION
**Pull request by johnsonshi**
Adds annotated diagnostic and error messages for various
commands. These annotations print the OSM namespace
and mesh name, which improves the CLI UX. Fixes #2455.

Signed-off-by: Johnson Shi <Johnson.Shi@microsoft.com>

**Description**:

Adds annotated diagnostic and error messages for various
commands. These annotations print the OSM namespace
and mesh name, which improves the CLI UX. Fixes #2455.

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [X]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Demo                   [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? No. If so, did you notify the maintainers and provide attribution? N/A.

### CODE REVIEW

**Comment by johnsonshi** [comment]
Opening this draft PR to check in with other OSM maintainers RE my proposed changes.

Before/After. Planning to add error message annotations to other subcommands as well.
image

**Comment by shashankram** [comment]
> Opening this draft PR to check in with other OSM maintainers RE my proposed changes.
> 
> Before/After. Planning to add error message annotations to other subcommands as well.
> image

This looks good, maybe add a new line after the error message and before the note :-)

**Comment by johnsonshi** [comment]
@shashankram // @delqn I think this should be ready now (added tests as well)

File: cmd/cli/dashboard.go [hunk 1]
```diff
@@ -81,15 +80,15 @@ func (d *dashboardCmd) run() error {
 		Get(context.TODO(), grafanaServiceName, metav1.GetOptions{})
 
 	if err != nil {
-		return errors.Errorf("Failed to get OSM Grafana service data: %s", err)
+		return annotateErrorMessageWithOsmNamespaceMessage("Failed to get OSM Grafana service data: %s", err)
```
Comment (shashankram) [inline cmd/cli/dashboard.go:83]:
```suggestion
		return annotateErrorMessageWithOsmNamespace("Failed to get OSM Grafana service data: %s", err)
```

**Comment by shashankram** [comment]
> @shashankram // @delqn I think this should be ready now (added tests as well)

Looks good, mark it ready for review so we can approve.

**Review by draychev — APPROVED** [review]
Thank you @johnsonshi 
Love adding more context around the output of the CLI commands!

**Comment by johnsonshi** [comment]
@shashankram This should be ready now :)

**Review by shashankram — APPROVED** [review]

**Review by michelleN — APPROVED** [review]

**Pull request merged**